### PR TITLE
Feature/refactoring loss fanctions

### PR DESCRIPTION
**Pull request by Yuto-web**
## 変更点
- loss関数のパラメータを設定できるように変更
- loss関数はncc/losses/loss_functions.pyにまとめて記述
## 問題点
- 特になし
## 備考
- 動作確認済み
- 学習未確認

### CODE REVIEW

File: farmer/domain/model/trainer_model.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ class Trainer(Config, ImageLoader, Optuna):
     cosine_lr_max: int = 0.01
     cosine_lr_min: int = 0.001
     optuna: bool = False
+    loss_params: Dict[str, float] = field(default_factory=dict)
```
[the remark this answers is not in the document]
Comment (Yuto-web) [reply farmer/domain/model/trainer_model.py]: 承知しました。

File: farmer/domain/tasks/build_model_task.py [hunk 2]
```diff
@@ -1,22 +1,15 @@
...
 from tensorflow import keras
 
-segmentation_models.set_framework('tf.keras')
 
-# loss functions
-cce_dice_loss = categorical_crossentropy + dice_loss
-cce_jaccard_loss = categorical_crossentropy + jaccard_loss
-categorical_focal_dice_loss = categorical_focal_loss + dice_loss
-categorical_focal_jaccard_loss = categorical_focal_loss + jaccard_loss
+segmentation_models.set_framework('tf.keras')
```
[the remark this answers is not in the document]
Comment (Yuto-web) [reply farmer/domain/tasks/build_model_task.py]: model作成に影響するかなと思ったのですが、関係ないみたいですね。
指摘箇所を削除して動作確認、学習確認済み
